src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
Sounds like the two should be the same. I'd run it by @davidfowl since this was originally his suggestion.